pkg/arvo/neo/cod/std/src/fil/messenger.md

```diff
@@ -0,0 +1,557 @@
+  --
+--
+```
+
+## The +deps arm
+The only part of this system that uses the `+deps` arm is `/message-sub`.
+
+```hoon
+::
+::  define dependencies
+++  deps
+  ::  deps:neo is (map term fief:neo)
+  %-  ~(gas by *deps:neo)
+  :~  :*  %pub
+          ::  fief:neo is [required=? =quay]
+          req=&
+          ::  quay:neo is (pair lash:neo (unit port:neo)
+          ::  lash:neo defines the shrub's state and pokes
+          [[%pro %sig] (sy %sig ~)]
+          %-  some
+          ::  port:neo constrains the shrub's kids
+          :-  %y
+          %-  ~(gas by *lads:neo)
+          :~  :-  [[%.n %da] |]
+              [[%pro %message] ~]
+          ==
+      ==
+  ==
+```
+
+With regards to the lifecycle of the shrub, the `+deps` arm types the shrubs whose names and locations are passed in as an argument in the `%make` card when this shrub is created.
+
+```hoon
+[%make %message-sub ~ (malt ~[[%pub (snoc host.poke %pub)]])]
+```
+
+In this card, `+malt` creates a `(map term pith:neo)` where the `term` is `%pub` and the path represented by the `pith` is `/path/to/chat/pub`. At `/path/to/chat/pub`, there’s a shrub which is the canonical “publisher” or “host” of a chat, whether that’s a group chat or a DM. Whenever there’s a state change in the publisher shrub, the `/message-sub` created in this card will be notified about it, but we’ll cover that in more detail in the next section.
+
+The `%pub` `term` will act as the key in the `map` of dependencies. This means `/message-sub` could have several dependencies. `%pub` doesn’t refer to anything, it’s just being defined in the `+deps` arm to differentiate between shrubs being passed in the card’s `conf:neo`; think of how you’d use head tags to route different cases in the `+on-poke` arm of a Gall agent.
+
+The `required` flag specfies whether the shrub being described in `+deps` needs to already exist in order for this shrub to build. If no publisher exists at `/path/to/chat/pub`, the `/message-sub` we’re trying to `%make` here will fail to build. It can only exist with reference to the publisher shrub.
+
+The last “new” idea here is `%sig`. If you look at `/imp/sig.hoon`, it’s just a `~` stub like the `%txt` implementation. `%sig` imps are not special and are not treated differently to any other stub, they’re just a stylistic convention to say that we don’t care about the state of the shrub in question; it could be anything, we won’t constrain it at all. There’s also a `/pro/sig.hoon` which lets us do the same thing for pokes.
+
+## Listening for state changes in our dependencies
+Unlike a Gall agent, a shrub does not send out `%facts` to subscribers in the event of changes to its state, all of which has to be manually implemented by that agent’s developer. Instead, the shrub listens for changes to its own state and automatically sends a `%rely` poke to shrubs that have declared it as a dependency. The developer of the listener shrub is the only one who has to write the logic to handle this.
```

Comment:
"Instead, the shrub listens for changes to its own state" not sure this is how I'd phrase it; %rely pokes are auto-sent when it's state changes